script.js

```diff
@@ -1,3 +1,83 @@
+// Vanilla JS debounce function, by Josh W. Comeau:
+// https://www.joshwcomeau.com/snippets/javascript/debounce/
+function debounce(callback, wait) {
+  let timeoutId = null;
+  return (...args) => {
+    window.clearTimeout(timeoutId);
+    timeoutId = window.setTimeout(() => {
+      callback.apply(null, args);
+    }, wait);
+  };
+}
+
+// Define variables for search field
+let searchForm, searchInput, searchClearButton;
+let searchFormFilledClassName = "search-has-value";
+
+// Clear the search input, and then return focus to it
+function clearSearchInput() {
+  searchForm.classList.remove(searchFormFilledClassName);
+  searchInput.value = "";
+  searchInput.focus();
+}
+
+// Have the search input and clear button respond 
+// when someone presses the escape key, per:
+// https://twitter.com/adambsilver/status/1152452833234554880
+function clearSearchInputOnKeypress(event) {
+  const searchInputDeleteKeys = ["Delete", "Escape"];
+  if (searchInputDeleteKeys.includes(event.key)) {
+    clearSearchInput();
+  }
+}
+
+// Ensure there's always a label for the search clear button,
+// just in case the localized label isn't available.
+// This label is a big part of what makes the custom button accessible.
+function getClearSearchButtonLabel() {
+  return window.searchClearButtonLabelLocalized === undefined || window.searchClearButtonLabelLocalized === "" ? "Clear search" : window.searchClearButtonLabelLocalized;
```

Comment:
We should be able to trust the api so no need for the fallback string.

Reply:
@luis-almeida I thought this might be worthwhile to have, in case something happens to the API, down the line. Does that make sense?

Reply:
I understand what you mean but this should never be the case.
If ourselves can't trust the API is stable than we can't ask our customers to trust us either.

Reply:
Can we remove this fallback logic?
This encourages distrust in the api and it's not something we should allow.